samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -132,9 +201,27 @@ void handleContainerLaunchFail(String processorId, String containerId, String pr
   }
 
   /**
-   * Handles an expired resource request for both active and standby containers. Since a preferred host cannot be obtained
-   * this method checks the availability of surplus ANY_HOST resources and launches the container if available. Otherwise
-   * issues an ANY_HOST request. Only applies to HOST_AFFINITY enabled cases
+   * Handles the state update on successful launch of a container, if this launch is due to a container placement action updates the
+   * related metadata to report success
+   *
+   * @param processorId logical processor id of container 0,1,2
+   */
+  void handleContainerLaunchSuccess(String processorId) {
+    if (hasActiveContainerPlacementAction(processorId)) {
+      ContainerPlacementMetadata metadata = getPlacementActionMetadata(processorId).get();
+      metadata.setActionStatus(ContainerPlacementMessage.StatusCode.SUCCEEDED,
+              "Successfully completed the container placement action");
+      LOG.info("Marking the container placement action success {}", metadata);
+    }
+  }
```

Comment:
This method is called from a single place, please inline with caller in ContainerProcessManager